documentation/src/docs/wire-protocol.md

```diff
@@ -7,106 +7,103 @@ You may obtain a copy of the License at
+4. `delegationToken`: String (2 byte) followed by that many bytes of Java's Modified UTF-8. This was added to perform _auth_. It is an opaque-to-the-client token provided by the Controller that says it's allowed to make this call.
 
-Appending
----------
+## Append Setup - Reply
 
-### Setup Append - Request
+1.  `RequestId`: Long (8 bytes). This field contains the client-generated ID that has been propagated to identify a client request.
+2.  `Segment`: String (2 bytes) followed by that many bytes of Java's Modified UTF-8). This Segment indicates the Stream Segment to append.
+3.  `writerId`: UUID (16 bytes). It identifies the requesting appender.
+4.  `lastEventNumber`: Long (8 bytes). It specifies the last event number in the Stream.
 
-1.  ConnectionId - UUID (16 bytes) Identifies this appender.
-2.  Segment to append to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
+## AppendBlock - Request
 
-### Append Setup - Reply
+1. `writerId`: UUID (16 bytes). It identifies the requesting appender.
+2. `Data`: This holds the contents of the block.
 
-1.  Segment that can be appended to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  ConnectionId - UUID (16 bytes) Identifies the requesting appender.
-3.  ConnectionOffsetAckLevel - Long (8 bytes) What was the last offset received and stored on this segment for this connectionId (0 if new)
+## AppendBlockEnd - Request
 
-### BeginAppendBlock - Request 
+1. `writerId`: UUID (16 bytes). It identifies the requesting appender.
+2. `sizeOfWholeEvents`: Integer (4 bytes). It is the total number of bytes in this block (starting from the beginning) that is composed of whole (meaning non-partial) events.
+3. `Data`: This holds the contents of the block.
+4. `numEvents`: Integer (4 bytes). It specifies the current number of events.
+5. `lastEventNumber`: Long (8 bytes). It specifies the value of last event number in the Stream.
 
-Only valid after SetupAppend has already been done successfully.
 
-1.  ConnectionId - UUID (16 bytes)
-2.  ConnectionOffset - Long (8 bytes) Data written so far over this connection to this segment
-3.  Length of data before EndAppendBlock message - Integer (4 bytes) 
+## Partial Event - Request/Reply
 
-### EndAppendBlock- Request
+-  **Data**: A Partial Event is an Event at the end of an Append block that did not fully fit in the Append block. The remainder of the Event will be available in the `AppendBlockEnd`.
 
-1.  ConnectionId - UUID (16 bytes)
-2.  ConnectionOffset - Long (8 bytes) Data written so far over this connection
-3.  Block Length - (4 Bytes) Total size of the block that was written. (Note this may more or less than the number of bytes between the BeginAppendBlock and this message)
+## Event - Request
 
-### Event - Request
+Only valid inside the block.
 
-Only valid inside of a block
+1.  `Data`: It contains the Event's data (not hold the contents of the block).
```

Comment:
You don't need to say "(not hold the contents of the block)" this just adds confusion. The event is inside the block. hence "Only valid inside the block"

Reply:
@tom Accepted suggestion